### PR TITLE
Feature/add upload api dotnet documentation

### PR DESCRIPTION
**Pull request by Roblinde**

### CODE REVIEW

File: docs/net/tutorials/management-api.md [hunk 1]
```diff
@@ -678,7 +678,89 @@ await client.UnarchiveAssetAsync("<new_asset_id>", version);
 To permanently delete an asset.
 
 ~~~csharp
-cawait lient.DeleteAssetAsync("<new_asset_id>", version);
+await client.DeleteAssetAsync("<new_asset_id>", version);
+~~~
+
+## Uploading files directly
+
+When creating assets in the example above your file needs to already be reachable on a url. That is what you put in the `UploadUrl` property. At times you want to upload a file directly from disk or
```
Comment (floelhoeffel) [inline docs/net/tutorials/management-api.md]: I would rephrase this paragraph. Sounds a bit cumbersome as is. Maybe dont start with explaining non direct file upload?

File: docs/net/tutorials/management-api.md [hunk 2]
```diff
@@ -678,7 +678,89 @@ await client.UnarchiveAssetAsync("<new_asset_id>", version);
...
+
+First upload your binary file.
+
+~~~csharp
+var binaryFileByteArray = File.ReadAllBytes("c:\example\yourfile.txt");
+var uploadReference = await client.UploadFileAsync("<new_asset_id>", version);
+~~~
+
+This returns an `UploadReference` that can then be used when creating an asset. You need to scrub a few properties from the `SystemProperties` of the reference. This is because these properties are
+not allowed when creating assets.
```
Comment (floelhoeffel) [inline docs/net/tutorials/management-api.md]: remove new line?

File: docs/net/tutorials/management-api.md [hunk 3]
```diff
@@ -678,7 +678,89 @@ await client.UnarchiveAssetAsync("<new_asset_id>", version);
...
+
+~~~csharp
+
+//Scrub the properties that are not allowed.
```
Comment (floelhoeffel) [inline docs/net/tutorials/management-api.md]: is "scrub" commonly used for nulling?

File: docs/net/tutorials/management-api.md (hunk 1 above)
Comment (Roblinde) [inline docs/net/tutorials/management-api.md]: Agree, removing that part entirely.
